src/System.Private.CoreLib/src/System/Decimal.cs

```diff
@@ -624,16 +611,16 @@ internal static void GetBytes(Decimal d, byte[] buffer)
 
         // Returns the larger of two Decimal values.
         //
-        internal static Decimal Max(Decimal d1, Decimal d2)
+        internal static ref Decimal Max(ref Decimal d1, ref Decimal d2)
         {
-            return Compare(d1, d2) >= 0 ? d1 : d2;
+            return ref DecCalc.VarDecCmp(ref d1, ref d2) >= 0 ? ref d1 : ref d2;
```

Comment:
`VarDecCmp` will trash the input values when it takes the `Scale factors are not equal` path in VarDecCmpSub. 

This may be a good use for the new C# `in` feature. It is like `ref` except the value is read-only. It would be it explicit what can change vs. not change.

Reply:
It used to trash the input values, but I changed it to use local variables (the perf impact was negligible). I verified the new behaviour using `in` parameters, but left them as `ref` because property access on an `in` parameter uses a local copy for non-readonly structs and `Decimal` is not yet readonly.

We had a discussion about marking `Decimal` readonly some time ago and I made this change with that in mind. But overall I think it might be easier to make it readonly after merging with CoreCLR.

Reply:
Sounds good. I have noticed that VarDecCmpSub was changed.